File: sdkwork-backend-sdk-java/src/main/java/com/sdkwork/backend/api/VipPackageApi.java
package com.sdkwork.backend.api;

import com.sdkwork.backend.http.HttpClient;
import com.sdkwork.backend.model.*;
import java.util.List;
import java.util.Map;

public class VipPackageApi {
    private final HttpClient client;
    
    public VipPackageApi(HttpClient client) {
        this.client = client;
    }

    /** Update VIP Package */
    public PlusApiResultPlusVipPackVO update(PlusVipPackForm body) throws Exception {
        return (PlusApiResultPlusVipPackVO) client.put(ApiPaths.backendPath("/vip/pack"), body);
    }

    /** Create VIP Package */
    public PlusApiResultPlusVipPackVO create(PlusVipPackForm body) throws Exception {
        return (PlusApiResultPlusVipPackVO) client.post(ApiPaths.backendPath("/vip/pack"), body);
    }

    /** Get VIP Packages by Page */
    public PlusApiResultPagePlusVipPackVO listByPage(QueryListForm body, Map<String, Object> params) throws Exception {
        return (PlusApiResultPagePlusVipPackVO) client.post(ApiPaths.backendPath("/vip/pack/list"), body, params);
    }

    /** Get All VIP Packages */
    public PlusApiResultListPlusVipPackVO listAllEntities(QueryListForm body) throws Exception {
        return (PlusApiResultListPlusVipPackVO) client.post(ApiPaths.backendPath("/vip/pack/list/all"), body);
    }

    /** Get VIP Package by ID */
    public PlusApiResultPlusVipPackVO getById(String id) throws Exception {
        return (PlusApiResultPlusVipPackVO) client.get(ApiPaths.backendPath("/vip/pack/" + id + ""));
    }

    /** Delete VIP Package */
    public PlusApiResultBoolean delete(String id) throws Exception {
        return (PlusApiResultBoolean) client.delete(ApiPaths.backendPath("/vip/pack/" + id + ""));
    }
}

File: sdkwork-backend-sdk-kotlin/src/main/kotlin/com/sdkwork/backend/api/VipPackageApi.kt
package com.sdkwork.backend.api

import com.sdkwork.backend.*
import com.sdkwork.backend.http.HttpClient

class VipPackageApi(private val client: HttpClient) {

    /** Update VIP Package */
    suspend fun update(body: PlusVipPackForm): PlusApiResultPlusVipPackVO? {
        return client.put(ApiPaths.backendPath("/vip/pack"), body) as? PlusApiResultPlusVipPackVO
    }

    /** Create VIP Package */
    suspend fun create(body: PlusVipPackForm): PlusApiResultPlusVipPackVO? {
        return client.post(ApiPaths.backendPath("/vip/pack"), body) as? PlusApiResultPlusVipPackVO
    }

    /** Get VIP Packages by Page */
    suspend fun listByPage(body: QueryListForm? = null, params: Map<String, Any>? = null): PlusApiResultPagePlusVipPackVO? {
        return client.post(ApiPaths.backendPath("/vip/pack/list"), body, params) as? PlusApiResultPagePlusVipPackVO
    }

    /** Get All VIP Packages */
    suspend fun listAllEntities(body: QueryListForm? = null): PlusApiResultListPlusVipPackVO? {
        return client.post(ApiPaths.backendPath("/vip/pack/list/all"), body) as? PlusApiResultListPlusVipPackVO
    }

    /** Get VIP Package by ID */
    suspend fun getById(id: String): PlusApiResultPlusVipPackVO? {
        return client.get(ApiPaths.backendPath("/vip/pack/$id")) as? PlusApiResultPlusVipPackVO
    }

    /** Delete VIP Package */
    suspend fun delete(id: String): PlusApiResultBoolean? {
        return client.delete(ApiPaths.backendPath("/vip/pack/$id")) as? PlusApiResultBoolean
    }
}
